cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,491 @@ Cask Data Application Platform Release Notes
    :backlinks: none
    :depth: 2
 
+`Release 4.1.0 <http://docs.cask.co/cdap/4.1.0/index.html>`__
+=============================================================
+
+New Features
+------------
+
+- :cask-issue:`CDAP-4556` - CDAP now uses environment variables in the ``spark-env.sh`` and
+  properties in the ``spark-defaults.conf`` when launching Spark programs.
+
+- :cask-issue:`CDAP-7596` - Added the ability to reuse an existing file system location
+  and Hive table when creating a partitioned file set.
+
+- :cask-issue:`CDAP-7597` - Added configuring the CDAP Explore database and table name for
+  a dataset using dataset properties.
+
+- :cask-issue:`CDAP-7676` - Added a prototype implementation for a rolling HDFS log
+  appender.
+
+- :cask-issue:`CDAP-7683` - Added a tool that pre-builds and loads the HBase coprocessors
+  required by CDAP onto HDFS.
+
+- :cask-issue:`CDAP-7685` - Provided SPI hooks that users can implement for performing
+  HBase DDL operations.
+
+- :cask-issue:`CDAP-8021` - Added the management of preferences at the application and
+  program levels.
+
+- :cask-issue:`CDAP-8025` - Added a tool to check a cluster's replication status.
```

Comment:
Can we create sections in the new Features section? To my understanding, the main new features in 4.1 are:
1. Revamped Log System
2. Replication, resiliency, and other zero-downtime features
3. Dataset improvements
4. Secure Impersonation and Authorization
5. Brand new UI

It would be really helpful if we can group these features in the above categories, otherwise this seems like a very big list to consume.